addons/website_forum/controllers/main.py

```diff
@@ -700,6 +700,4 @@ def convert_answer_to_comment(self, forum, post, **kwarg):
 
     @http.route('/forum/<model("forum.forum"):forum>/post/<model("forum.post"):post>/comment/<model("mail.message"):comment>/delete', type='json', auth="user", website=True)
     def delete_comment(self, forum, post, comment, **kwarg):
-        if not request.session.uid:
-            return {'error': 'anonymous_user'}
-        return post.unlink_comment(comment.id)[0]
+        return post.unlink_comment(comment.id)
```

Comment:
Feels like we should add a check access on post which would add some value to this unlink comment method. A bit strange actually.

Reply:
There are some checks in `unlink_comment` but yes the responsibility split is... original.

Reply:
(also I removed the uid check because the controller is `auth=user` so there's always a uid)